otelcol/command_components.go

```diff
@@ -109,6 +115,18 @@ func newComponentsCommand(set CollectorSettings) *cobra.Command {
 				})
 			}
 			components.BuildInfo = set.BuildInfo
+
+			confmapProviderFactories := set.ConfigProviderSettings.ResolverSettings.ProviderFactories
+			for _, confmapProvider := range confmapProviderFactories {
+				provider := confmapProvider.Create(set.ConfigProviderSettings.ResolverSettings.ProviderSettings)
+				scheme := provider.Scheme()
+				module := set.ProviderModules[scheme+"provider"]
```

Comment:
Using the scheme as the name for the provider feels like it's not guaranteed to always work, I'm a bit concerned some providers may have a difference in the future even if all upstream providers have matching names and schemes.

Could we do the following instead:
1. In the builder, set up all the metadata we need given the Go module metadata. We should probably just do all the transformations we need here.
2. Loop through `set.providerModules` and retrieve this data directly.

What do you think?

Reply:
Thinking on it more, we should call out the scheme instead of or in addition to the module name since the scheme is the part users actually care about. So we'll need to still read from `confmapProviderFactories` to get the scheme.

However I still don't think we can guarantee that future providers won't have names like `acmecorpsecretsprovider` that read URIs with schemes like `acs://corp.my/secret`. We could potentially use reflection to find the module for a given provider and match them up, but I'm not sure about the best way to do this given the provider interfaces don't give a module name.

Reply:
I see! Would that make sense to extend the provider interface so as to be able to provide the package or module name?

Reply:
Unfortunately we likely can't expand the interface any further now that confmap is stable. However it looks like we can do something like this in the map inside main.go.tmpl:

```
{{.Name}}.NewFactory().Create(confmap.ProviderSettings{}).Scheme()
```

If we key on the scheme like this and have the go module as the value, I think we only need a few changes inside the main.go template.

Reply:
That works, indeed! Thank's for the suggestion!